core/lib/spree/permission_sets/user_management.rb

```diff
@@ -2,7 +2,16 @@ module Spree
   module PermissionSets
     class UserManagement < PermissionSets::Base
       def activate!
-        can :manage, Spree.user_class
+        can [:admin, :display, :create, :update], Spree.user_class
+
+        # due to how cancancan filters by associations,
+        # we have to define this twice, once for `accessible_by`
+        can :update_email, Spree.user_class, spree_roles: { id: nil }
+        # and once for `can?`
+        can :update_email, Spree.user_class do |user|
+          user.spree_roles.none?
```

Comment:
@Senjai I'm not sure I follow this is about ability to edit email, not ability to change roles. Are you saying that you suggest to stores that only SuperUsers can assign roles? That still doesn't quite cover all of the potential privilege escalation, e.g. viewing orders -> editing orders.